libs/wingsdk/src/shared/sandbox.ts

```diff
@@ -92,33 +95,45 @@ export class Sandbox {
     return context;
   }
 
-  private async loadBundleOnce() {
+  private async createBundle() {
     // load bundle into context on first run
-    if (this.loaded) {
-      return;
-    }
-
     const workdir = await mkdtemp(path.join(tmpdir(), "wing-bundles-"));
     const bundle = createBundle(this.entrypoint, [], workdir);
     this.entrypoint = bundle.entrypointPath;
 
-    const code = await readFile(this.entrypoint, "utf-8");
+    this.code = await readFile(this.entrypoint, "utf-8");
 
     if (process.env.DEBUG) {
-      const bundleSize = Buffer.byteLength(code, "utf-8");
+      const bundleSize = Buffer.byteLength(this.code, "utf-8");
       this.options.log?.(true, "log", `Bundled code (${bundleSize} bytes).`);
     }
+  }
+
+  private loadBundleOnce() {
+    if (this.loaded) {
+      return;
+    }
+
+    if (!this.code) {
+      throw new Error("Bundle not created yet - please report this as a bug");
+    }
 
     // this will add stuff to the "exports" object within our context
-    vm.runInContext(code, this.context, {
+    vm.runInContext(this.code!, this.context, {
       filename: this.entrypoint,
     });
 
     this.loaded = true;
   }
 
   public async call(fn: string, ...args: any[]): Promise<any> {
-    await this.loadBundleOnce();
+    // wait for the bundle to finish creation
+    await this.createBundlePromise;
+
+    // load the bundle into context on the first run
```

Comment:
Very good call.

So basically you implemented lambda provisioned concurrency ;-)